cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-rococo/src/tests/snowbridge.rs

```diff
@@ -226,33 +226,42 @@ fn register_token() {
 
 #[test]
 fn send_token_to_penpal() {
-	let asset_hub_sovereign = BridgeHubRococo::sovereign_account_id_of(
-		MultiLocation {
-			parents: 1,
-			interior: X1(Parachain(ASSETHUB_PARA_ID)),
-		}
-	);
-	BridgeHubRococo::fund_accounts(vec![(
-		asset_hub_sovereign.clone(),
-		INITIAL_FUND,
-	)]);
-
-	// Fund ethereum sovereign in asset hub
-	AssetHubRococo::fund_accounts(vec![
-		(AssetHubRococoReceiver::get(), INITIAL_FUND),
-	]);
+	let asset_hub_sovereign = BridgeHubRococo::sovereign_account_id_of(MultiLocation {
+		parents: 1,
+		interior: X1(Parachain(ASSETHUB_PARA_ID)),
+	});
+	BridgeHubRococo::fund_accounts(vec![(asset_hub_sovereign.clone(), INITIAL_FUND)]);
 
 	PenpalA::fund_accounts(vec![
 		(PenpalAReceiver::get(), INITIAL_FUND),
 		(PenpalASender::get(), INITIAL_FUND),
 	]);
 
-	let weth_asset_id: MultiLocation = (
-		Parent,
-		Parent,
-		Ethereum { chain_id: 15 },
-		AccountKey20 { network: None, key: WETH }
-	).into();
+	let weth_asset_location: MultiLocation =
+		(Parent, Parent, Ethereum { chain_id: 15 }, AccountKey20 { network: None, key: WETH })
+			.into();
+	let weth_asset_id = weth_asset_location.into();
+
+	let origin_location =
+		MultiLocation { parents: 2, interior: weth_asset_location.interior.split_last().0 };
+
+	// Fund ethereum sovereign in asset hub
+	let ethereum_sovereign: AccountId =
+		hex!("da4d66c3651dc151264eee5460493210338e41a7bbfca91a520e438daf180bf5").into();
+	AssetHubRococo::fund_accounts(vec![(ethereum_sovereign.clone(), INITIAL_FUND)]);
+	// Create asset on assethub.
+	AssetHubRococo::execute_with(|| {
+		assert_ok!(<AssetHubRococo as AssetHubRococoPallet>::ForeignAssets::create(
+			pallet_xcm::Origin::Xcm(origin_location).into(),
+			weth_asset_id,
+			asset_hub_sovereign.clone().into(),
```

Comment:
The `admin` for our assets on rococo should be the snowbridge sovereign account. Infact the fact that this test works is concerning because technically only Snowbridge sovereign should be able to issue new WETH tokens.

Reply:
Looked into this and it `issuer` is only checked when called from the `mint` extrinsic. It seems xcm-executor has access to mint without checking `issuer`. But this is covered by the `IsReserve` config item on the xcm-executor which is configured to only allow the bridged locations.